src/Data/Lens/Fold.purs

```diff
@@ -148,6 +149,15 @@ toListOfOn s p = toListOf p s
 
 infixl 8 toListOfOn as ^..
 
+-- | Collects the foci of a `Fold` into an array.
+toArrayOf :: forall s t a b. Fold (Endo (->) (Array a)) s t a b -> s -> Array a
+toArrayOf p = foldrOf p Array.cons [] 
```

Comment:
This is going to be quadratic in the length of the array being constructed. I think `Array.fromFoldable <<< toListOf p` would be a better implementation. (There may well be a better one still, though.)

Reply:
ok I can make that change

Reply:
alright it's updated